### ISSUE TITLE
[Resolved] USB-C to DisplayPort not working

### ISSUE THREAD

**Issue by rpruden** [ZombieTheBest/XPS9350-10.13-HighSierra#23]
I decided to update to your fork so that I can run High Sierra. I was originally running the build that Syscl put together and everything was working pretty well. The new High Sierra setup is also working pretty well, though I can not connect my external DisplayPort monitor anymore. I get no display detected.

I am using a single cable that connects from the USB-C port on the Laptop to the DisplayPort on the monitor. This worked just fine previously. Any thoughts as to what might be the problem? I am new around here so please pardon me if I didn't provide enough detail.

Oh also, on another note, After running the install.sh file, I can no longer open my config.plist in Clover Configuration Editor, it crashes the program every time now. Is that normal?

**Comment by ggctseng** [comment]
@rpruden I know we're discussing various items on my thread, but I note I had the same config.plist issue with it and Clover Config and I believe there may be issues with the config.plist--though I haven't looked at the Deploy.sh to see what changes it is supposed to make, if any, to the plist.  What I did was a more conservative approach, which is to copy entries using Clover Config from the config.plist generated by Deploy.sh and adding sections to my "base" config.plist (the one by Rehabman for the HD540).

Your monitor issues may be related.

**Comment by rpruden** [comment]
@ggctseng I am pretty sure that the Deploy.sh does make changes to the default config.plist. I like your idea of manually merging the two files. I will probably try that when I have some time. I uploaded a copy of mine on the other tread, so hopefully you can help me see what I'm either missing or erroneously adding.

I'd really like to know which line added by deploy.sh is breaking the config.plist so that we could let @ZombieTheBest know and have a fix implemented. I wonder if it could also just be a bug in the current version of Clover Configurator.

**Comment by ggctseng** [comment]
@rpruden Just FYI, I downloaded the config.plist you provided in the other thread.  Was that the one in there the one crashing your CC?  I was able to open it in Clover Configurator with no errors.

**Comment by ggctseng** [comment]
@rpruden 
@ZombieTheBest (my question for you is at the very bottom but the details are below).

As I noted in my other boot to installer issue thread, the DisplayPort on my U3818DW display (USB-C<-->USB-C) XPS would hotplug when I had the "bare minimum" config.plist from rehabman that allowed me to install High Sierra.

I then ran deploy.sh, but still used that same bare minimum config.plist and confirmed that the hotplug still worked.  However, after I moved entries over from the config.plist modified by deploy.sh, my U3818DW display does not hotplug anymore--I am aware of the issue related to USB-C hotplugging for devices, but obviously the DisplayPort hotplugging was working.

Of course, when I boot with U3818DW attached: the display, as well as USB devices connected to the U3818DW works.

When I go back and boot with a backup copy of the bare minimum config.plist I used before I moved over any entries from the config.plist from the deploy.sh, the display hotplugs fine.

Can you let me know what I should modify in the config.plist to re-enable hotplugging for DP on the USB-C port? I don't care about the USB device hotplugging and can wait for a resolution.

**Comment by rpruden** [comment]
@ggctseng 
> Just FYI, I downloaded the config.plist you provided in the other thread. Was that the one in there the one crashing your CC? I was able to open it in Clover Configurator with no errors.

Yes, that is the same one. I had noticed once that it failed to load in CC on one of my user profiles, and was successful on another. It seems as though it's been somewhat inconsistent if it will crash it.

**Comment by rpruden** [comment]
@ggctseng I have tried going back to the "vanilla" version of the config.plist without running the deploy.sh in order to test if my USB-C/DisplayPort would work as expected, but this actually didn't work for me. I used the config.plist that is distributed in this repo. That is the same as Rehabman's config right?

I'm really not sure what is up with this and why I can't get my external monitor to function like it used to. What version of BIOS are you on? I checked mine and it's 1.4.17. Also, what version does your SMBIOS indicate? I am using MacbookPro13,3 as mentioned before, but the default config.plist tries to use the MacbookPro13,2 (and so did my Sierra installation). If I try to use the 13,2 my system just reboots a few seconds after the Apple logo appears.

**Comment by ggctseng** [comment]
@rpruden Sorry, was busy today. DO NOT USE MacBookPro13,3 because that's for the XPS 15", which has a 2nd GPU in addition to the Iris HD540.  I think this explains why I had issues identified with the U3818DW monitor. I had thought my issues with the monitor were related to where I put the kexts, etc., but it wasn't (see below).

tl;dr - use MacbookPro13,1

Here's the long explanation, while you're waiting for your MaXPS to reboot, haha:

Have you tried using MacbookPro13,1? This is what I am using now. If you look it up on everymac.com, MacbookPro13,1 is almost identical to our spec:

MacbookPro13,1
[https://everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-i7-2.4-13-late-2016-retina-display-no-touch-bar-specs.html](https://everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-i7-2.4-13-late-2016-retina-display-no-touch-bar-specs.html)

MacbookPro13,2 (has Touch Bar *AND* 4 Thunderbolt ports *AND* a 3.3 GHz i7 (<-- I think one of these is causing your crash).
[https://everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-i7-3.3-13-late-2016-retina-display-touch-bar-specs.html](https://everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-i7-3.3-13-late-2016-retina-display-touch-bar-specs.html)

Clover Configurator is not accurate in its description of MacbookPro13,1, which is why I ended up doing what you did to use MacbookPro13,3, which is for the 15" MacbookPro. In general, it seems better to not use Clover Configurator to mess with your config.plist, although it is useful to view it and grab information such as bootlog. HOWEVER,  you can and should use it to generate the SMBIOS information after you boot into MacbookPro13,1--just save that config.plist as a separate file (I saved it as config-SMBIOS.plist) and copy the information you need over to your config.plist.

BTW, I decided to go with HFS+ based on the recommendation of Rehabman (I'm sure not in a position to argue with him, lol), so I ended up reinstalling High Sierra.  I started a thread here if you want to follow my process:

[https://tonymacx86.com/threads/issues-with-display-port-and-pd-over-usb-c-xps-13-9350.243891/#post-1676677](https://tonymacx86.com/threads/issues-with-display-port-and-pd-over-usb-c-xps-13-9350.243891/#post-1676677)

This time, I'm going to try to use hot patching and rely on Clover to do most of the heavy lifting.  Thus, I'm pretty much going to avoid running deploy.sh (unless I can't figure things out and get desperate, haha) but instead I will use information in the script.  We'll see how that goes.

Let me know if changing to MacbookPro13,1 works.

**Comment by rpruden** [comment]
@ggctseng WOW! I changed over to 13,1 and now my DisplayPort works perfectly just as before. I guess I should have taken a closer look at all of the specs before deciding on 13,3.

I've created a fresh config.plist using the source in this repo. I added the Fix_Headers option and generated a new SMBIOS. I then transferred my serial number in to the new config.plist and then ran deploy.sh again. Everything seems to be working pretty well.

The only problem I noticed was that after a re-boot, my bluetooth wasn't working, but this is an issue I had occasionally before I upgraded to High Sierra. It's one of those things that can be fixed by shutting down and powering back up and I was used to it already. It's interesting that it never happened when I was using 13,3. I wonder if something can be learned from this to fix these intermittent bluetooth issues.

Thank you so much for helping me figure this out. I was getting tired of the fuzziness of the USB video adapter I was using. My screen looks MUCH better now.

**Comment by ggctseng** [comment]
@rpruden Did you use Clover Config to add the Fix_Headers? I used deploy.sh to create a "base" config.plist to which I will be able to compare my edits (working on it more over the weekend). I noticed that the deploy.sh generated config.plist crashed Clover Config, so it's either in deploy.sh or Clover Config.

Re: bluetooth, so you're saying that 13,3=perfectly working BT?  Definitely interesting. Can you start another thread so maybe @ZombieTheBest will be aware and address?  I keep having to remind myself that the model designation DOES change how OS X operates, despite everyone saying what is in "About this Mac" is "irrelevant".

Other than this BT-issue, SD card, and (obviously) USB-C issues, do you see any other issues? I'm still getting ACPI  errors (during verbose boot and in console--I think related to plugging/unplugging power) and a weird battery capacity adjustment issue.  Also, audio doesn't redirect properly to monitor but works on the laptop (volume keys don't work, either).

Yeah, I'm not picky about the refresh freq. of the monitor, but having a clear image is definitely critical. Good we've gotten it working for you. We can probably mark this as solved.
